enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,499 @@
+
+When multiple namespaces share the same network, services will be reachable in between namespaces. It is also possible to select endpoints between both namespaces for these services.
+
+As previously mentioned, Kubernetes API and DNS services will be accessible by all pods.
+
+Endpoint slices will provide the IPs of the default network in Kubernetes API. For this implementation the required endpoints are those IP addresses which reside on the unique primary network. In order to solve this problem, OVN-Kubernetes may create its own endpoint slices or may choose to do dynamic lookups at runtime to map endpoints to their primary IP address. Leveraging a second set of endpoint slices will be the preferred method, as it creates less indirection and gives explicit Kube API access to what IP addresses are being used by OVN-Kubernetes.
+
+Kubelet healthchecks to pods are queried via the default network. When endpoints are considered unhealthy they will be removed from the endpoint slice, and thus their primary IP will be removed from the OVN load balancer. However, it is important to note that the healthcheck is being performed via the default interface on the pod, which ensures the application is alive, but does not confirm network connectivity of the primary interface. Therefore, there could be a situation where OVN networking on the primary interface is broken, but the default interface continues to work and reports 200 OK to Kubelet, thus rendering the pod serving in the endpoint slice, but unable to function. Although this is an unlikely scenario, it is good to document.
+
+
+### Network Policy
+
+Configuring network policies that allow traffic between unique network namespaces will not be invalidated or ignored by OVN-Kubernetes. If a user configures a network policy that would have no effect, (like stopping namespace A from talking to B on different networks) then OVN-Kubernetes will still configure this policy in OVN. These policies will also have no effect between namespaces that are assigned to different networks as there will be no network connectivity between the namespaces. However, there may still be a desire to configure network policy between namespaces that share the same network, as well as for other reasons like ingress/egress control.
+
+
+### Risks and Mitigations
+
+The biggest risk with this feature is hitting scale limitations. With many namespaces and networks, the number of internal OVN objects will multiply, as well as internal kernel devices, rules, VRFs. There will need to be a large-scale effort to determine how many networks we can comfortably support.
```

Comment:
I'd also add risk to break the integration / compatibility with existing projects.